site-src/guides/implementers-guide.md

```diff
@@ -0,0 +1,367 @@
+then it's likely that your Kubernetes client code will return serious errors.
+Tips to deal with this are also detailed below.
+
+The CRD definitions for Gateway API objects all contain two specific
+annotations:
+
+- `gateway.networking.k8s.io/bundle-version: <semver-release-version>`
+- `gateway.networking.k8s.io/channel: <channel-name>`
+
+The concepts of "bundle version" and "channel" (short for "release channel") are
+explained in our [versioning][versioning] documentation.
+
+Implementations may use these to determine what schema versions are installed in
+the cluster, if any.
+
+[versioning]: /concepts/versioning
+
+### Changes to the Gateway API CRDs are backwards compatible
+
+Part of the contract for Gateway API CRDs is that changes _within an API version_
+must be _compatible_.
+
+"Within an API Version" means changes to a CRD that occur while the same API version
+(`v1alpha2` or `v1` for example) is in use, and "compatible" means that any new
+fields, values, or validation will be added to ensure that _previous_
+objects _will still be valid objects_ after the change.
+
+This means that once Gateway API objects move to the `v1` API version, then _all_
+changes must be compatible.
+
+This contract also means that an implementation will not fail with a higher version
+of the API than the version it was written with, because the newer schema being
+stored by Kubernetes will definitely be able to be serialized into the older version
+used in code by the implementation.
+
+Similarly, if an implementation was written with a _higher_ version, the newer
+values that it understands will simply _never be used_, as they are not present
+in the older version.
+
+A similar guarantee occurs between the "experimental" and "standard" channels for
+objects in the same API version, so an implementation may be written with the
+experimental API definitions, but work just fine with having only the standard
+definitions installed - there will be fields or values that will never be used.
+The same applies for an implementation written using the standard API definitions
+running in a cluster with the experimental definitions installed.
+
+## Implementation Rules and Guidelines
+
+### CRD Management
+
+For a Gateway API implementation to work, the Gateway API CRDs must be installed
+in the Kubernetes cluster the implementation is watching.
+
+Implementations have two options: automatically installing CRDs or requiring
+installation before working. Both have tradeoffs.
+
+Either way has certain things that SHOULD be true, however:
+
+Whatever method is used, cluster admins SHOULD attempt to ensure that
+the Bundle version of the CRDs is not _downgraded_. Although we ensure that
+API changes are backwards compatible, changing CRD definitions can change the
+storage version of the resource, which could have unforseen effects. Most of the
+time, things will probably work, but if it doesn't work, it will most likely 
+break in weird ways.
```

Comment:
Probably worth noting somewhere in here that older versions of the API may be missing fields or features which could be very disruptive for users.